Source/BugsnagFileStore.m

```diff
@@ -128,13 +128,17 @@ - (NSUInteger)fileCount {
 }
 
 - (NSArray *)allFiles {
+    return [[self allFilesByName] allValues];
+}
+
+- (NSDictionary <NSString *, NSDictionary *> *)allFilesByName {
     NSArray *fileIds = [self fileIds];
-    NSMutableArray *files =
-            [NSMutableArray arrayWithCapacity:[fileIds count]];
+    NSMutableDictionary *files =
+    [NSMutableDictionary dictionaryWithCapacity:[fileIds count]];
     for (NSString *fileId in fileIds) {
         NSDictionary *fileContents = [self fileWithId:fileId];
         if (fileContents != nil) {
-            [files addObject:fileContents];
+            [files setObject:fileContents forKey:fileId];
```

Comment:
AppCode warns that modern Obj-C syntax can be used here: `files[fileId] = fileContents`;